services/datalad/datalad_service/handlers/upload.py

```diff
@@ -12,6 +12,16 @@
 from datalad_service.common.annex import CommitInfo
 
 
+def move_files(upload_path, dataset_path):
+    print(dataset_path)
+    for filename in pathlib.Path(upload_path).glob('**/*'):
+        if os.path.isfile(filename):
+            target = os.path.join(dataset_path, os.path.relpath(
+                filename, start=upload_path))
+            os.makedirs(target, exist_ok=True)
```

Comment:
I think you want to create the parent dir?

```suggestion
            pathlib.Path(target).parent.mkdir(parents=True, exist_ok=True)
```

Reply:
Yep, thanks. I fixed the tests to check that these are files and not just directories with the right names.